lib/shared/serialization.js

```diff
@@ -0,0 +1,53 @@
+/**
+ * @fileoverview Object serialization utils.
+ * @author Bryan Mishkin
+ */
+
+"use strict";
+
+/**
+ * Check if a value is a primitive or plain object created by the Object constructor.
+ * @param {any} val the value to check
+ * @returns {boolean} true if so
+ * @private
+ */
+function isSerializablePrimitiveOrPlainObject(val) {
+    return (
+        val === null ||
+        typeof val === "string" ||
+         typeof val === "boolean" ||
+         typeof val === "number" ||
+         (typeof val === "object" && val.constructor === Object) ||
+         Array.isArray(val)
+    );
+}
+
+/**
+ * Check if an object is serializable.
+ * Functions or objects like RegExp cannot be serialized by JSON.stringify().
+ * Inspired by: https://stackoverflow.com/questions/30579940/reliable-way-to-check-if-objects-is-serializable-in-javascript
+ * @param {any} obj the object
+ * @returns {boolean} true if the object is serializable
+ */
+function isSerializable(obj) {
+    if (!isSerializablePrimitiveOrPlainObject(obj)) {
+        return false;
+    }
+    for (const property in obj) {
```

Comment:
Does "object" mean that this function expects only objects? If that's not the case, then it might be better to rename `obj` to `val` to clarify that it accepts any value, and also do for-in only on objects so that it doesn't, for example, iterate over string characters (currently, for string test cases, it calls `isSerializablePrimitiveOrPlainObject` for each character of the string).

Reply:
Good point, changed to `val`, and only looping for objects.